### PR TITLE
Move slip10/bip32 logic from stronghold to crypto.rs

### PR DESCRIPTION
**Pull request by tensor-programming**
# Description of change

Moves existing Slip10 logic from stronghold into crypto.rs with minor API changes.  Slip10/bip32 requires `hmac`, `sha`, `ed25519` and `random` (for testing). 99% of the code follows the original code written by @rootmos 

Failed with clippy and testing because of unnecessary result in `tests/x25519.rs:17:1` but issue was resolved. 

## Links to any relevant issues

Be sure to reference any related issues by adding `fixes issue #`.

## Type of change

Choose a type of change, and delete any options that are not relevant.

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] Enhancement (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Fix

## How the change has been tested

Runs against the ed25519 test vectors. 

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by nothingismagick — APPROVED** [review]

**Comment by nothingismagick** [comment]
Thanks @tensor-programming. @Wollac could you also review this for us too?

File: src/slip10.rs [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+                    },
+                ],
+            },
+            // https://github.com/satoshilabs/slips/blob/master/slip-0010.md#test-vector-2-for-ed25519
+            TestVector {
+                seed: "fffcf9f6f3f0edeae7e4e1dedbd8d5d2cfccc9c6c3c0bdbab7b4b1aeaba8a5a29f9c999693908d8a8784817e7b7875726f6c696663605d5a5754514e4b484542",
+                master_chain_code: "ef70a74db9c3a5af931b5fe73ed8e1a53464133654fd55e7a66f8570b8e33c3b",
+                master_private_key: "171cb88b1b3c1db25add599712e36245d75bc65a1a5c9e18d76f9f2b1eab4012",
+                chains: vec![
+                    TestChain {
```
Comment (Wollac) [inline src/slip10.rs]: The testcase for `Chain::empty()` present at https://github.com/satoshilabs/slips/blob/master/slip-0010.md#test-vector-2-for-ed25519 is missing.
This is the only test missing from the entire specification so we should definitely also add it.

**Review by Wollac — CHANGES_REQUESTED** [review]
One tiny remark:
- Could we maybe move the tests from the specification to the `test` folder to be consistent with the other primitives?

Everything else looks good to me!

File: src/slip10.rs (hunk 1 above)
Comment (tensor-programming) [reply src/slip10.rs]: Alright, no problem. I will add this testcase.

**Comment by tensor-programming** [comment]
> One tiny remark:
> 
> * Could we maybe move the tests from the specification to the `test` folder to be consistent with the other primitives?
> 
> Everything else looks good to me!

Can do. 

Tests have been moved to a `tests/slip10.rs` file and the zero chain case for test vector 2 was added.

**Review by nothingismagick — CHANGES_REQUESTED** [review]
Can you please add a .changes file

**Review by Wollac — APPROVED** [review]

**Comment by tensor-programming** [comment]
> Can you please add a .changes file

Its done in the latest iteration.

**Review by nothingismagick — APPROVED** [review]

**Pull request merged**